JDP230502-25 - working implementation of UserController-User-UserMapp…

WorstWorm opened the pull request:
…er-UserDbService

 On branch JDP230502-25
 Changes to be committed:
	modified:   src/main/java/com/kodilla/ecommercee/controller/UserController.java
	modified:   src/main/java/com/kodilla/ecommercee/domain/User.java
	modified:   src/main/java/com/kodilla/ecommercee/mapper/UserMapper.java
	new file:   src/main/java/com/kodilla/ecommercee/service/UserDbService.java

WorstWorm committed:
JDP230502-25 - working implementation of UserController-User-UserMapper-UserDbService

 On branch JDP230502-25
 Changes to be committed:
	modified:   src/main/java/com/kodilla/ecommercee/controller/UserController.java
	modified:   src/main/java/com/kodilla/ecommercee/domain/User.java
	modified:   src/main/java/com/kodilla/ecommercee/mapper/UserMapper.java
	new file:   src/main/java/com/kodilla/ecommercee/service/UserDbService.java

sew-k commented on src/main/java/com/kodilla/ecommercee/controller/UserController.java:
```diff
@@ -1,100 +1,108 @@
-                userDtoList.add(userDtoIn);
-                return ResponseEntity.ok(userDtoIn);
-            }
+    @PutMapping(path="/updateUser/{id}", consumes= MediaType.APPLICATION_JSON_VALUE)
```
Why you use longer path with '/updateUser/' before {id} ?

WorstWorm replied:
Because on PUT I have also user blocking (/blockUser/{id}) and key generation (/generateKey/{id}).

WorstWorm committed:
JDP230502-25 - solved problem with automatic id change on updateUser

 On branch JDP230502-25
 Your branch is up to date with 'origin/JDP230502-25'.

 Changes to be committed:
	modified:   src/main/java/com/kodilla/ecommercee/controller/UserController.java
	modified:   src/main/java/com/kodilla/ecommercee/service/UserDbService.java

JakubFir commented:
My thoughts - remove all Logic from Controller move it to service, dont use Optional, do .OrEsleThrow() and make your own exception class UserNotFound

WorstWorm commented:
> My thoughts - remove all Logic from Controller move it to service, dont use Optional, do .OrEsleThrow() and make your own exception class UserNotFound

Most of the logic should be in the service - you're right. Here, however, I have another problem because I want to return appropriate HTTP responses. If I move this logic down to the service it will not be possible.
But I think your solution is more correct so I will follow your suggestion. I'll try to fix it today.

JakubFir commented:
Yes you can specify response, if you mean this for example
@ResponseStatus(code = HttpStatus.NOT_FOUND)
public class UserNotFoundException extends RuntimeException 
And then on Controller you leave responseStatus.Ok
And you orElseThrow(new usernotfoundexception) in service and if its bad request it returns not found

WorstWorm commented:
I am temporarily closing the pull request - I am making corrections agreed at the meeting and in correspondence.
